app/src/androidTest/java/ch/epfl/qedit/HomePopUpTest.java

```diff
@@ -0,0 +1,128 @@
+import ch.epfl.qedit.view.home.HomePopUp;
+import ch.epfl.qedit.view.home.HomeQuizListFragment;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class HomePopUpTest {
+    @Rule
+    public final IntentsTestRule<HomeActivity> rule =
+            new IntentsTestRule<>(HomeActivity.class, false, false);
+
+    @Test
+    public void testCreationOfPopUp() {
+        final User user = createUser();
+        launchActivity();
+        RecyclerView.Adapter adapter =
+                new HomeQuizListFragment().new CustomAdapter(rule.getActivity());
+
+        // Test if no exception is thrown
+        Assert.assertThat(
+                new HomePopUp(rule.getActivity(), user, adapter), instanceOf(HomePopUp.class));
+
+        finishActivity();
+    }
+
+    @Test
+    public void testEdit() {
+        final User user = createUser();
+        launchActivity();
+
+        rule.getActivity()
+                .runOnUiThread(
+                        new Runnable() {
+                            public void run() {
+                                final RecyclerView.Adapter adapter =
+                                        new HomeQuizListFragment()
+                                        .new CustomAdapter(rule.getActivity());
+
+                                final HomePopUp homePopUp =
+                                        new HomePopUp(rule.getActivity(), user, adapter);
+                                final AlertDialog alertDialog = homePopUp.popUpEdit("", 1);
+                                alertDialog.show();
+
+                                // Check it exists
+                                onView(withText("Cancel"));
+                                onView(withText("Done"));
+
+                                alertDialog.cancel();
+                            }
+                        });
+
+        finishActivity();
+    }
+
+    @Test
+    public void testDelete() {
+        final User user = launchActivity();
+        final RecyclerView.Adapter adapter =
+                new HomeQuizListFragment().new CustomAdapter(rule.getActivity());
+        new Handler(Looper.getMainLooper())
+                .post(
+                        new Runnable() {
+                            @Override
+                            public void run() {
+
+                                HomePopUp homePopUp =
+                                        new HomePopUp(rule.getActivity(), user, adapter);
+                                final AlertDialog alertDialog = homePopUp.popUpWarningDelete("", 2);
+                                alertDialog.show();
+                                // Check it exists
+                                onView(withText("Cancel"));
+                                onView(withText("Yes"));
+                                alertDialog.cancel();
+                            }
+                        });
+
+        finishActivity();
+        //        final RecyclerView.Adapter adapter =
```

Comment:
Can we remove those kinds of comments from PRs ready to be merged ? They don't help the code's readability -_-

Reply:
Sure, yes